new-tutorial/3_message_passing.py

```diff
@@ -0,0 +1,353 @@
+"""
+Write your own GNN module
+=========================
+
+This tutorial assumes that you already know :doc:`the basics of training a
+GNN for node classification <1_introduction>`. However,
+sometimes you would like to invent a new way of aggregating neighbor
+information or a new way of using the aggregation.
+
+By the end of this tutorial you will be able to
+
+-  Understand DGL’s message passing APIs.
+-  Implement GraphSAGE convolution module by your own.
+
+"""
+
+import dgl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+######################################################################
+# This tutorial assumes that you already know the pipeline of full-graph
+# node classification. If not, please refer to the
+# :doc:`introduction <1_introduction>`.
+# 
+# The following code for data loading and training loop is directly copied
+# from the introduction tutorial.
+# 
```

Comment:
I will first talk about how to use message passing APIs and then add this training loop section as FYI.